pkg/collector/py/loader.go

```diff
@@ -292,13 +278,39 @@ func expvarPy3Warnings() interface{} {
 	return py3Warnings
 }
 
-func addExpvarPy3Warnings(checkName string, warnings []string) (float64, string) {
+func addExpvarPy3Warnings(checkName string, checkFilePath string) {
 	statsLock.Lock()
 	defer statsLock.Unlock()
 
-	if len(warnings) > 0 {
-		py3Warnings[checkName] = warnings
-		return a7NotReady, a7TagNotReady
+	status := a7TagUnknown
+	metricValue := 0.0
+	if checkFilePath != "" {
+		// __file__ return the .pyc file path
+		if strings.HasSuffix(checkFilePath, ".pyc") {
+			checkFilePath = checkFilePath[:len(checkFilePath)-1]
+		}
+
+		if warnings, err := validatePython3(checkName, checkFilePath); err != nil {
+			status = a7TagUnknown
+		} else if len(warnings) == 0 {
+			status = a7TagReady
+			metricValue = 1.0
+		} else {
+			status = a7TagNotReady
+		}
+	}
+	py3Warnings[checkName] = status
+
+	// add a serie to the aggregator to be sent on every flush
+	tags := []string{
+		fmt.Sprintf("status:%s", status),
+		fmt.Sprintf("check_name:%s", checkName),
 	}
-	return a7Ready, a7TagReady
+	tags = append(tags, agentVersionTags...)
+	aggregator.AddExtraSeries(&metrics.Serie{
```

Comment:
in an AD scenario where the Agent constantly schedules and unschedules checks:
* we can assume that the number of the different integrations that are loaded is relatively small and bounded (i.e. over its entire runtime the Agent will never load more than a small number of different integrations), but
* the number of individual checks that will be scheduled/unscheduled is not small and not bounded (and on top of that, I _think_ the AD logic always "loads" the integration when a new check config is scheduled)

and this leads me to think that we can't just append series to the list of additional series that are sent at every flush, we should make sure we only send at most one series per integration name.

Let me know what you think